gateway/middleware.go

```diff
@@ -474,6 +474,10 @@ func (t BaseMiddleware) ApplyPolicies(session *user.SessionState) error {
 			tags[tag] = true
 		}
 
+		for k, v := range policy.MetaData {
+			session.MetaData[k] = v
```

Comment:
Let's ensure that key can override metadata if needed. So first check if it has non-empty value.

Reply:
On each update, `ApplyPolicies` is used. So, we need to keep a separate field to hold overrides. Another thing, `tags` also works like this. If user wants to change something he should go to policy to make change. Therefore, it is better for us to keep it consistent.

Reply:
I see, makes sense!